### PR TITLE
Fix enrichments preview

### PR DESCRIPTION
**Pull request by guilbill**
Fix https://trello.com/c/kAMofyrc/73-preview-de-ce-qui-est-envoy%C3%A9-aux-enrichers

- Add tests for enrichment preview
- Fix ini template for multiple values

https://user-images.githubusercontent.com/560852/148239107-082aee9a-14be-451f-98fa-37c8cedaebc8.mp4

### CODE REVIEW

File: src/api/services/enrichment/directPathMultipleValues.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ plugin = analytics
 
 [assign]
 path = value
-value = get("value.[[SOURCE COLUMN]]").castArray().map(JSON.parse).reduce((acc, item) => item)
+value = get("value").update("[[SOURCE COLUMN]]", (item) => { try { return JSON.parse(item)} catch {return item } }).get("[[SOURCE COLUMN]]")
```
Comment (arimet) [inline src/api/services/enrichment/directPathMultipleValues.txt]: How did you find this solution ? 😄

File: src/api/services/enrichment/enrichment.js [hunk 2]
```diff
@@ -10,8 +10,10 @@ import { ENRICHING, PENDING } from '../../../common/progressStatus';
 import { jobLogger } from '../../workers/tools';
 
 const getSourceData = async (ctx, sourceColumn) => {
-    const excerpt = await ctx.dataset.getExcerpt();
-    const sourceData = excerpt.find(line => !!line[sourceColumn])[sourceColumn];
+    const excerpt = await ctx.dataset.getExcerpt({
```
Comment (floo51) [inline src/api/services/enrichment/enrichment.js]: Is there a reason it's named excerpt here and excerptLines further down?

File: src/api/services/enrichment/enrichment.js [hunk 3]
```diff
@@ -77,13 +81,14 @@ export const getEnrichmentDataPreview = async ctx => {
 };
 
 const isBasicType = sourceData => {
-    return typeof sourceData === 'string' || typeof sourceData === 'number';
+    return typeof sourceData !== 'object';
```
Comment (floo51) [inline src/api/services/enrichment/enrichment.js]: What about date types? Do we actually handle them? typeof a date is 'object' too

File: src/api/services/enrichment/directPathMultipleValues.txt (hunk 1 above)
Comment (floo51) [reply src/api/services/enrichment/directPathMultipleValues.txt]: Next dev on the project will be like.... 
image

Can we leave an explanation somewhere?

**Review by floo51 — APPROVED** [review]

File: src/api/services/enrichment/enrichment.js (hunk 3 above)
Comment (guilbill) [reply src/api/services/enrichment/enrichment.js]: Date are serialized as String, so I think it's good.

**Review by floo51 — APPROVED** [review]

**Pull request merged**